source/common/stream_info/utility.cc

```diff
@@ -5,180 +5,44 @@
 namespace Envoy {
 namespace StreamInfo {
 
-const std::string ResponseFlagUtils::NONE = "-";
-const std::string ResponseFlagUtils::DOWNSTREAM_CONNECTION_TERMINATION = "DC";
-const std::string ResponseFlagUtils::FAILED_LOCAL_HEALTH_CHECK = "LH";
-const std::string ResponseFlagUtils::NO_HEALTHY_UPSTREAM = "UH";
-const std::string ResponseFlagUtils::UPSTREAM_REQUEST_TIMEOUT = "UT";
-const std::string ResponseFlagUtils::LOCAL_RESET = "LR";
-const std::string ResponseFlagUtils::UPSTREAM_REMOTE_RESET = "UR";
-const std::string ResponseFlagUtils::UPSTREAM_CONNECTION_FAILURE = "UF";
-const std::string ResponseFlagUtils::UPSTREAM_CONNECTION_TERMINATION = "UC";
-const std::string ResponseFlagUtils::UPSTREAM_OVERFLOW = "UO";
-const std::string ResponseFlagUtils::UPSTREAM_RETRY_LIMIT_EXCEEDED = "URX";
-const std::string ResponseFlagUtils::NO_ROUTE_FOUND = "NR";
-const std::string ResponseFlagUtils::DELAY_INJECTED = "DI";
-const std::string ResponseFlagUtils::FAULT_INJECTED = "FI";
-const std::string ResponseFlagUtils::RATE_LIMITED = "RL";
-const std::string ResponseFlagUtils::UNAUTHORIZED_EXTERNAL_SERVICE = "UAEX";
-const std::string ResponseFlagUtils::RATELIMIT_SERVICE_ERROR = "RLSE";
-const std::string ResponseFlagUtils::STREAM_IDLE_TIMEOUT = "SI";
-const std::string ResponseFlagUtils::INVALID_ENVOY_REQUEST_HEADERS = "IH";
-const std::string ResponseFlagUtils::DOWNSTREAM_PROTOCOL_ERROR = "DPE";
-const std::string ResponseFlagUtils::UPSTREAM_MAX_STREAM_DURATION_REACHED = "UMSDR";
-const std::string ResponseFlagUtils::RESPONSE_FROM_CACHE_FILTER = "RFCF";
-const std::string ResponseFlagUtils::NO_FILTER_CONFIG_FOUND = "NFCF";
-const std::string ResponseFlagUtils::DURATION_TIMEOUT = "DT";
-const std::string ResponseFlagUtils::UPSTREAM_PROTOCOL_ERROR = "UPE";
-const std::string ResponseFlagUtils::NO_CLUSTER_FOUND = "NC";
-
-void ResponseFlagUtils::appendString(std::string& result, const std::string& append) {
+void ResponseFlagUtils::appendString(std::string& result, absl::string_view append) {
   if (result.empty()) {
     result = append;
   } else {
-    result += "," + append;
+    result += ",";
+    result += append;
```

Comment:
perf nit: `absl::StrCat(&result, ",", append);` will right-size the buffer in result, where as separately appending twice may cause 2 resizes.

Reply:
adopt the below tech